pkg/controllers/default_resources.go

```diff
@@ -48,23 +52,60 @@ func getReleaseImagePullSpec() string {
 
 }
 
-func ScaffoldHostedCluster(hyd *hypdeployment.HypershiftDeployment) *hyp.HostedCluster {
-	hostedCluster := &hyp.HostedCluster{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      hyd.Name,
-			Namespace: helper.GetHostingNamespace(hyd),
-			Annotations: map[string]string{
-				constant.AnnoHypershiftDeployment: fmt.Sprintf("%s/%s", hyd.Namespace, hyd.Name),
-			},
-		},
-		Spec: *hyd.Spec.HostedClusterSpec,
-	}
+func (r *HypershiftDeploymentReconciler) scaffoldHostedCluster(ctx context.Context, hyd *hypdeployment.HypershiftDeployment) (*unstructured.Unstructured, error) {
+	hostedCluster := &unstructured.Unstructured{}
+	hostedCluster.SetAPIVersion(hyp.GroupVersion.String())
+	hostedCluster.SetKind("HostedCluster")
+	hostedCluster.SetName(hyd.Name)
+	hostedCluster.SetNamespace(helper.GetHostingNamespace(hyd))
+	hostedCluster.SetAnnotations(map[string]string{
+		constant.AnnoHypershiftDeployment: fmt.Sprintf("%s/%s", hyd.Namespace, hyd.Name),
+	})
+
+	if !hyd.Spec.Infrastructure.Configure && len(hyd.Spec.HostedClusterRef.Name) != 0 {
+		hcRef := hyd.Spec.HostedClusterRef
+
+		gvr := schema.GroupVersionResource{
+			Group:    "hypershift.openshift.io",
+			Version:  "v1alpha1",
+			Resource: "hostedclusters",
+		}
+		unstructHostedCluster, err := r.DynamicClient.Resource(gvr).Namespace(hyd.Namespace).Get(ctx, hcRef.Name, v1.GetOptions{})
+		if err != nil {
+			_ = r.updateStatusConditionsOnChange(hyd, hypdeployment.PlatformConfigured, metav1.ConditionFalse,
+				fmt.Sprintf("HostedClusterRef %v:%v not found", hyd.Namespace, hcRef.Name), hypdeployment.MisConfiguredReason)
+
+			errMsg := fmt.Sprintf("failed to get HostedClusterRef: %v:%v", hyd.Namespace, hcRef.Name)
+			r.Log.Error(err, errMsg)
+			return nil, fmt.Errorf(errMsg)
+		}
 
-	// Pass all appropriate annotations to the HostedCluster
-	// Find Annotation references here: https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go
-	transferHostedClusterAnnotations(hyd.Annotations, hostedCluster.Annotations)
+		// Validate hosted cluster by converting the unstructured HC to the concrete HC obj
+		hostedClusterRef := &hyp.HostedCluster{}
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(unstructHostedCluster.UnstructuredContent(), hostedClusterRef)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to convert unstructured hosted cluster to concrete type: %v:%v", hyd.Namespace, hcRef.Name)
+			r.Log.Error(err, errMsg)
```

Comment:
We aren't swallowing the errors in the return, so this code is double logging, since its putting an error into the log and then into the return, and return from the reconcile eventually is logged as well. Also is your goal to cause a quick reconcile with the error, or is it better to log it and not reconcile again, till the user makes a change?

The precedence in the other parts of the code is to return the condition update error, so the user sees the condition and can make a change to fix the problem, vs us endlessly looping because the Get failed.